src/lib/components/EventQAndA/index.js

```diff
@@ -0,0 +1,88 @@
+import {BaseElement} from '../BaseElement';
+import './_styles.scss';
+
+/**
+ * Filters out Q&A's based on `data-category` property.
+ *
+ * @extends {BaseElement}
+ * @final
+ */
+class EventQAndA extends BaseElement {
+  constructor() {
+    super();
+    this.selectCategory = this.selectCategory.bind(this);
+
+    this.catageories = new Set();
+    this.categorySelectElement = document.createElement('select');
+    this.categorySelectElement.classList.add('w-select--borderless');
+    this.categorySelectElement.addEventListener('change', this.selectCategory);
+  }
+  connectedCallback() {
+    super.connectedCallback();
+    this.addDropDown();
+
+    this.querySelectorAll('[data-category]').forEach((element) => {
+      element.addEventListener('click', this.closeDetail);
+    });
+  }
+
+  disconnectedCallback() {
+    this.categorySelectElement.removeEventListener(
```

Comment:
you add this handler in the constructor but then remove in disconnected.

We manage the subordinate element's lifecycle, you don't need this remove here.